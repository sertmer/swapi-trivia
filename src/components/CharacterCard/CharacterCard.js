import React, { Component }from 'react'
import './CharacterCard.scss'
import PropTypes from 'prop-types'

class CharacterCard extends Component {
  constructor(props) {
    super(props) 
      this.state = {
      isFavorite: false
    }
  }

  favoriteThis = () => {
    console.log('props:', this.props)
    let character = {
      name: this.props.name,
      isFavorite: true,
      homeworld: this.props.homeworld,
      population: this.props.population,
      species: this.props.species,
      films: this.props.films,
      unFavoriteThis: this.props.unFavoriteThis
    }
    console.log('character', character)
    this.props.updateFavorite(character)
  }

  

  render() {
  return (
    <article className='character_card'>
    <h2> {this.props.name} </h2>
    <p> Homeworld: {this.props.homeworld[0]} </p>
    <p> Population: {this.props.homeworld[1]} </p>
    <p> Species: {this.props.species} </p>
    <p> Related Films: {this.props.films} </p>
    { this.props.isFavorite ? 
    <button onClick={() => this.props.unFavoriteThis(this.props.name)}>Unfavorite</button> :
      <button onClick={this.favoriteThis}>Favorite</button>}
    </article>
    )
  }
}

export default CharacterCard;

CharacterCard.propTypes = {
  name: PropTypes.string.isRequired,
  homeworld: PropTypes.array.isRequired,
  species: PropTypes.string.isRequired,
  films: PropTypes.array.isRequired
}