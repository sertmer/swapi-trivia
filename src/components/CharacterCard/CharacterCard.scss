.character_card {
  border: 1px solid #f9d71c;
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 2%;
  width: 25%;
  border-radius: 5px;
  height: 400px;
  justify-content: space-around;
  background-color: #1a1a1a;

  p {
    margin: 2%;
  }
}