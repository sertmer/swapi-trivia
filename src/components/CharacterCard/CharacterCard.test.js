import React from 'react'
import CharacterCard from './CharacterCard'
import { shallow } from 'enzyme'

describe('CharacterCard', () => {
  let wrapper
  let mockProps = {
    className: 'character_card',
    key: 1,
    name: 'bob',
    homeworld: 'tatooine',
    population: 2,
    species: 'droid',
    films: []
  }

  beforeEach(() => {
    wrapper = shallow(<CharacterCard {...mockProps}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})