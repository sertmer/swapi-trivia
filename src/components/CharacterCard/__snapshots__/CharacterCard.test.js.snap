// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot 1`] = `
<article
  className="character_card"
>
  <h2>
     
    bob
     
  </h2>
  <p>
     Homeworld: 
    t
     
  </p>
  <p>
     Population: 
    a
     
  </p>
  <p>
     Species: 
    droid
     
  </p>
  <p>
     Related Films: 
     
  </p>
</article>
`;
